### PR TITLE
feat: Crossplane Kubernetes Provider

### LINKED ISSUE #1133: Crossplane kubernetes provider
Adds the Crossplane Kubernetes Provider to the crossplane addon.

### PR DESCRIPTION
**Pull request by bobdoah**
### What does this PR do?

Adds the Crossplane Kubernetes Provider to the crossplane addon. 

### Motivation

I am intending to use this provider myself. It would be helpful to have it easily toggled as part of the addons.

Resolves #1133 

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [x] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [x] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: docs/add-ons/crossplane.md [hunk 1]
```diff
@@ -67,4 +68,15 @@ crossplane_jet_aws_provider = {
...
 ```
 
+_NOTE: Crossplane requires cluster-admin permissions to create and update Kubernetes resources._
+
+Config to deploy [Kubernetes provider](https://github.com/crossplane-contrib/provider-kubernetes)
+```hcl
+# Creates ProviderConfig -> kubernetes-provider
+crossplane_kubernetes_provider = {
+  enable                   = true
+  provider_kubernetes_version     = "main"  # Get the latest version from  https://github.com/crossplane-contrib/provider-jet-aws
```
Comment (askulkarni2) [inline docs/add-ons/crossplane.md:78]: Fix url and pin to specific version.

```suggestion
  provider_kubernetes_version     = "v0.5.0"  # Get the latest version from  https://github.com/crossplane-contrib/provider-kubernetes
```

File: examples/crossplane/main.tf [hunk 2]
```diff
@@ -97,6 +97,13 @@ module "eks_blueprints_kubernetes_addons" {
     additional_irsa_policies = ["arn:aws:iam::aws:policy/AmazonS3FullAccess"]
   }
 
+  # Creates ProviderConfig -> kbuernetes-provider
+  kubernetes_provider = {
+    # NOTE: Crossplane requires cluster-admin permissions to create and update resources.
+    enable                      = true
+    provider_kubernetes_version = "main"
```
Comment (askulkarni2) [inline examples/crossplane/main.tf:104]:
```suggestion
    provider_kubernetes_version = "v0.5.0"
```

**Review by askulkarni2 — COMMENTED** [review]
@bobdoah thanks or the PR! Please follow community guidelines to open an issue first. 

Some minor comments, otherwise looking good to me. There are some CI issues that need to be addressed.

@vara-bonthu can you please review?

File: docs/add-ons/crossplane.md (hunk 1 above)
Comment (bobdoah) [reply docs/add-ons/crossplane.md:78]: I did initially try with "v0.5.0" but that version doesn't appear to exist in DockerHub: https://hub.docker.com/r/crossplane/provider-kubernetes/tags. The provider was failing to run. I also tried v0.5.0 from the upbound repository, but that was similarly failing to run: https://marketplace.upbound.io/providers/crossplane-contrib/provider-kubernetes/v0.5.0.

**Comment by bobdoah** [comment]
> @bobdoah thanks or the PR! Please follow community guidelines to open an issue first.
> 
> Some minor comments, otherwise looking good to me. There are some CI issues that need to be addressed.
> 
> @vara-bonthu can you please review?

I've pinned it at v.0.4.1 and (hopefully) fixed the example issue.

**Review by askulkarni2 — APPROVED** [review]

**Pull request merged**